### PR TITLE
Populate Help menu with weblinks to docs

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
# Description
This is a stab at fixing #5094 
For now, the Help menu is populated with weblinks to:
Getting started, Tutorials, Examples Gallery, and napari.org
While it might be nice to open a widget or something similar to provide help & documentation *within* napari, I think this is a good start to making that info more accessible from within napari. In VS Code the Help menu also has a `Documentation` item that just opens the default web browser to the docs page, so a similar approach using the standard module `webbrowser` was used here.
Also by populating the menu, the `napari info` menu item now uses `QAction.AboutRole` to properly place it the `napari` menu on macOS.
image
image

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
#5094 

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [x] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by Czaki** [comment]
Hm. What about linking to give nersion docs? So instead stable use tag?

**Comment by psobolewskiPhD** [comment]
> Hm. What about linking to give nersion docs? So instead stable use tag?

So that the menu link connects to the docs of the specific version? 
That would be nice. I wanted to link to release notes, but wasn't sure how to do the version check of napari. I will poke around some more.

**Comment by Czaki** [comment]
stable docs: https://napari.org/stable/index.html
0.4.16 docs: https://napari.org/0.4.16/index.html
development docs: https://napari.org/dev/index.html

**Comment by psobolewskiPhD** [comment]
> stable docs: https://napari.org/stable/index.html
> 
> 0.4.16 docs: https://napari.org/0.4.16/index.html
> 
> development docs: https://napari.org/dev/index.html

Right, but is PRing the menu link every release the right idea? To keep the link synced with version. 

Edit: old napari won't have the items in the menu so it's just say 0.5.0 linking to own docs after 0.5.x is released.

**Comment by Czaki** [comment]
Do you mean about something like this to check the current napari version form the code?

```python
In [1]: from packaging.version import parse

In [2]: from napari import __version__

In [3]: v = parse(__version__)

In [4]: v.is_devrelease
Out[4]: True

In [5]: str(v)
Out[5]: '0.4.16rc2.dev296+g27a87e027'

In [6]: "dev" if v.is_devrelease else str(v)
Out[5]: 'dev'
```

**Comment by psobolewskiPhD** [comment]
> Do you mean about something like this to check the current napari version form the code?
> 
> 
> 
> ```python
> 
> In [1]: from packaging.version import parse
> 
> 
> 
> In [2]: from napari import __version__
> 
> 
> 
> In [3]: v = parse(__version__)
> 
> 
> 
> In [4]: v.is_devrelease
> 
> Out[4]: True
> 
> 
> 
> In [5]: str(v)
> 
> Out[5]: '0.4.16rc2.dev296+g27a87e027'
> 
> 
> 
> In [6]: "dev" if v.is_devrelease else str(v)
> 
> Out[5]: 'dev'
> 
> ```

Yea. I think options are:
1. Use stable. Easy, but if someone doesn't update their napari then docs may not fully match. 
2. Use release and update the link at every new release. Easy but somewhat annoying?
3. Use something like your code above to keep the links matched to the napari version automatically.

**Comment by Czaki** [comment]
As napari does not provide stable API some plugin creators does not support all napari version and on this phase of development it is expected to people will use one or two versions in back.

**Comment by psobolewskiPhD** [comment]
> As napari does not provide stable API some plugin creators does not support all napari version and on this phase of development it is expected to people will use one or two versions in back. 

Ok, so options 2 or 3 then.

**Comment by psobolewskiPhD** [comment]
@Czaki using your code example with parse works for me locally on both 0.4.16 and `main` (dev). Seems like the best solution!

**Comment by psobolewskiPhD** [comment]
I swapped `napari info` to the bottom of the menu after a divider—this is typical on Windows, though the name is `About ...` not `... info`. Plus, I think it makes sense to have the Help at the top. I also added the Release notes, if the version isn't `dev`. I think easy access to the release notes is handy, plus it seems typical in other software (e.g. VS Code)
One more thing that could be added, that Chrome has:
`Report an issue` that links to GitHub issues
Alternately, there could be a link to https://forum.image.sc/tag/napari or zulip...
Going to CC @melissawm here for any thoughts on whether something else or different links should be used.

**Comment by brisvag** [comment]
Awesome work @psobolewskiPhD! I agree on the versioning of the link. However, it would also be great if we had a nice colorful warning on the website that tells people that that's not the latest version (see for example the [numpy docs](https://numpy.org/doc/1.19/index.html)). Of course this shouldn't be part of this PR, it's just tangentially related :)

**Comment by psobolewskiPhD** [comment]
Great Idea! It would also ensure someone reaching docs from Help knows that they are running outdated napari.
@brisvag BTW can you post a screenshot from non-macOS?

**Comment by brisvag** [comment]
Of the menu?

image

**Comment by psobolewskiPhD** [comment]
Of the contents of Help 🤣

**Comment by Czaki** [comment]
obraz

**Comment by psobolewskiPhD** [comment]
Thanks @Czaki Guessing you're on dev branch? (No release notes shown).
I think it looks good to me.

**Comment by Czaki** [comment]
I do screenshots from `Populate_Help` branch. Do you ask about screenshots from the current code?

**Comment by psobolewskiPhD** [comment]
No, no, sorry I wasn't clear and was also obviously stupid—of course you're testing this PR branch not something else.
Anyhow, I was just glad to see the check for dev/stable worked so release notes arnt shown in the menu. I've tested locally by modifying the file in `site-packages` of 0.4.16.

**Comment by Czaki** [comment]
obraz

I prefer to edit `__version__` in `napari/_version.py`

**Comment by liu-ziyang** [comment]
FYI There may be some merge conflicts depending on merge order because of some refactor planned to move all menus to the app model, this is the one for the help menu: https://github.com/napari/napari/pull/4922

**Review by goanpeca — APPROVED** [review]
Love this @psobolewskiPhD ! Thanks for working on it 🚀

**Review by melissawm — APPROVED** [review]
Looks great to me, thanks @psobolewskiPhD ! The only suggestion I might give is adding an icon to make explicit that these are links to external resources (something like this: https://fontawesome.com/icons/up-right-from-square?s=solid&f=classic) but this should probably go through a design review, so looks good for now!

I also took the liberty of opening an issue for the old version banners: #5146

**Comment by psobolewskiPhD** [comment]
That could be cool, not sure how to accomplish that. Also, at least on macOS that's not common. E.g. this is the native Twitter app. The menu item opens a browser tab:
image
The behavior here should be similar—I'm not forcing a new window. Should we?

**Comment by melissawm** [comment]
I don't think that's a blocker - we should just put this in as is and maybe we can revisit that in the future.

File: napari/_qt/menus/help_menu.py
Comment (GenevieveBuckley) [inline napari/_qt/menus/help_menu.py:66]: Should the napari info stay at the top of the help menu? We ask people to submit it whenever they report bugs, so might be good to keep it easy to find

Comment (GenevieveBuckley) [reply napari/_qt/menus/help_menu.py:66]: > I swapped napari info to the bottom of the menu after a divider—this is typical on Windows, though the name is About ... not ... info. Plus, I think it makes sense to have the Help at the top. I also added the Release notes, if the version isn't dev. I think easy access to the release notes is handy, plus it seems typical in other software (e.g. VS Code)

Ah, I see that was a deliberate choice. Ok, so long as we tell people where to find it when they submit bug reports that should be fine.

**Comment by GenevieveBuckley** [comment]
> I'm not forcing a new window. Should we?

No, opening a new tab is a good way to go.

**Review by GenevieveBuckley — APPROVED** [review]
Looks good to me

**Comment by andy-sweet** [comment]
@psobolewskiPhD : this would great to get in soon! Anything blocking this other than the update with `main`? I'll let you go ahead and merge since this has core-dev approval already.

**Comment by psobolewskiPhD** [comment]
@andy-sweet Alas, I think this needs to be completely redone, due to the app-model migration, see:
https://github.com/napari/napari/pull/4922#issuecomment-1297695071

Edit: well not completely, since the content seems rather settled. But in terms of implementation. It's on my to-do list, but I've not had a chance to get to it. Maybe this weekend?
Making `draft` to be safe.

**Comment by psobolewskiPhD** [comment]
I re-did this PR due to the above mentioned app-model change.
Closing this in favor of https://github.com/napari/napari/pull/5399

**Pull request closed without merging**
